src/confluent_kafka/src/Admin.c

```diff
@@ -4712,6 +4813,15 @@ static void Admin_background_event_cb (rd_kafka_t *rk, rd_kafka_event_t *rkev,
                 break;
         }
 
+        case RD_KAFKA_EVENT_DELETERECORDS_RESULT:
+        {
+                const rd_kafka_DeleteRecords_result_t *c_delete_records_res = rd_kafka_event_DeleteRecords_result(rkev);
+                const rd_kafka_topic_partition_list_t *c_delete_records_res_list = rd_kafka_DeleteRecords_result_offsets(c_delete_records_res);
+                
+                result = c_parts_to_py(c_delete_records_res_list);
```

Comment:
I'm not sure the responses are returned in the same order as the requests, so that it's possible to use `_make_futmap_result_from_list`

Also we should use a `DeletedRecords` object as in Java, instead of a `TopicPartition` because that could be extended in future and in that case we cannot apply the same changes.

Reply:
We are converting it to the futmap in the end keyed at `TopicPartition` so there is no issue in that I think.

We can create `DeletedRecords` class as Java instead of directly using `offset` in the response.

Reply:
Created a new class DeleteRecords containing only offset and made changes according to that. Please review.